packages/next/next-server/lib/i18n/detect-locale-cookie.ts

```diff
@@ -0,0 +1,19 @@
+import { IncomingMessage } from 'http'
+import cookie from 'next/dist/compiled/cookie'
+
+export function detectLocaleCookie(req: IncomingMessage, locales: string[]) {
+  let detectedLocale: string | undefined
+
+  if (req.headers.cookie && req.headers.cookie.includes('NEXT_LOCALE')) {
+    const header = req.headers.cookie
+    const { NEXT_LOCALE } = cookie.parse(
```

Comment:
Could we add `req.cookies` support for pages as well so that this parsing is not needed (just using `req.cookies`)

Reply:
Updated to expose `req.cookies` for all pages instead of just API routes